pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1152,173 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func runningOnKVM() bool {
+	_, enabledHVs := hypervisor.GetAvailableHypervisors()
+	for _, enabledHV := range enabledHVs {
+		if enabledHV == hypervisor.KVMHypervisorName {
+			return true
+		}
+	}
+	return false
+}
+
+func getHostCpusNum() (int, error) {
+	if !runningOnKVM() {
+		log.Warnf("Running not on KVM/QEMU system, skip getting CPUs info")
+		return 0, nil
+	}
+	// The `nproc` util relies on the /proc/sys file system, that is not properly accessible from a container.
+	// So, have to run the command with access to the host FS.
+	out, err := exec.Command("chroot", "/hostfs", "nproc", "--all").Output()
+	if err != nil {
+		return -1, err
+	}
+	outStr := strings.TrimSpace(string(out))
+	maxHostCpus, err := strconv.Atoi(outStr)
+	if err != nil {
+		return -1, err
+	}
+	return maxHostCpus, nil
+}
+
+func setCgroupCpuset(vmUUIDAndVersion string, cpus string) error {
```

Comment:
Does [github.com/containerd/cgroups](https://pkg.go.dev/github.com/containerd/cgroups@v1.0.4) which we already use have this functionality? Better to avoid our own gcroup wrapper code if possible.

Reply:
I'll check it, ok.

Reply:
It works but complicates the code. But let it be. I'll add this approach to the code on the next iteration.

Reply:
OK

Reply:
Done